.github/actions/templates/publishModule/action.yml

```diff
@@ -75,3 +87,36 @@ runs:
         } else {
           Write-Output "Pipeline is configured to not publish to TemplateSpecs"
         }
+
+    - name: 'Publish module to private bicep registry'
+      shell: pwsh
+      # if: ${{ inputs.bicepRegistryDoPublish == 'true' }}
+      run: |
+        # Workaround until composite actions support conditions
+        # Once available, remove this if and enable the step condition
+        if('${{ inputs.bicepRegistryDoPublish }}' -eq 'true') {
+
+          # Load used functions
+          . "$env:GITHUB_ACTION_PATH/scripts/Publish-ModuleToPrivateBicepRegistry.ps1"
+
+          $templateFilePath = Join-Path "$env:GITHUB_WORKSPACE" "${{ inputs.templateFilePath }}"
+
+          # Extracts Microsoft.KeyVault/vaults from e.g. C:\arm\Microsoft.KeyVault\vaults\deploy.bicep
+          $moduleIdentifier = (Split-Path $templateFilePath -Parent).Replace('\','/').Split('/arm/')[1]
+
+          $functionInput = @{
+            moduleIdentifier    = $moduleIdentifier
+            templateFilePath    = $templateFilePath
+            bicepRegistryName   = '${{ inputs.bicepRegistryName }}'
+            bicepRegistryRgName = '${{ inputs.bicepRegistryRgName }}'
+            customVersion       = '${{ inputs.customVersion }}'
+            versioningOption    = '${{ inputs.versioningOption }}'
+          }
+
+          Write-Verbose "Invoke task with" -Verbose
+          Write-Verbose ($functionInput | ConvertTo-Json | Out-String) -Verbose
+
+          Publish-ModuleToPrivateBicepRegistry @functionInput -Verbose
+        } else {
+          Write-Output "Pipeline is configured to not publish to TemplateSpecs"
+        }
```

Comment:
Could we move this to a PowerShell file and just call it directly as a single command in run? i.e. main.ps1. If its in PowerShell we get better support in tools. Maybe this restructure could happen separately, but I'm more a fan of having code in file where we actually can get support from the tools we are using. It goes for all the places we have workflows backed by powershell (or other). As a reference. Look at the [Azure/Login action.yml](https://github.com/Azure/login/blob/master/action.yml) and [action/checkout](https://github.com/actions/checkout/blob/main/action.yml)

Reply:
Not sure how you would get the inputs from a PowerShell context however, but maybe it just works using `${{ input.something }}`-syntax in the PowerShell file.

Reply:
Opened another issue for this: #427